addons/point_of_sale/static/src/js/Popups/ClosePosPopup.js

```diff
@@ -152,16 +152,14 @@ class ClosePosPopup extends AbstractAwaitablePopup {
                 }
                 window.location = "/web#action=point_of_sale.action_client_pos_menu";
             } catch (error) {
-                const iError = identifyError(error);
-                if (
-                    iError instanceof ConnectionLostError ||
-                    iError instanceof ConnectionAbortedError
-                ) {
-                    await this.showPopup("ErrorPopup", {
-                        title: this.env._t("Network Error"),
-                        body: this.env._t("Cannot close the session when offline."),
-                    });
+                if (identifyError(error) instanceof ConnectionLostError) {
+                    // Cannot redirect to backend when offline, let error handlers show the offline popup
+                    // FIXME POSREF: doing this means closing again when online will redo the beginning of the method
+                    // although it's impossible to close again because this.closeSessionClicked isn't reset to false
+                    // The application state is corrupted.
+                    throw error;
                 } else {
+                    // FIXME POSREF: why are we catching errors here but not anywhere else in this method?
```

Comment:
`ConnectionAbortedError` is only raised when the connection was aborted by the client, which currently is done nowhere in the code base, but if it was, it would *not* mean that we are offline, just that the RPC was aborted, probably because the result of the ongoing RPC is already obsolete. In the case where an RPC can be aborted, the abort error should be caught specifically if we expect it.

I also find it dubious that we're trying to redirect the user to the backend when the connection was lost, presumably they are offline and the redirect will not work.

Reply:
> I also find it dubious that we're trying to redirect the user to the backend when the connection was lost, presumably they are offline and the redirect will not work.

Nah, we're only redirecting if there were some server errors that prevented us from closing the pos because the backend has a special flow for those cases (which was not yet implemented to the front). 

If the user lost connection, it was showing an errorPopup